simple_salesforce/util.py

```diff
@@ -34,4 +34,14 @@ def date_to_iso8601(date):
 
 class SalesforceError(Exception):
     """Base Salesforce API exception"""
-    pass
+
+    message = 'Unknown error occurred for {url}. Response content: {content}'
+
+    def __init__(self, url, status, resource_name, content):
+        self.url = url
+        self.status = status
+        self.resource_name = resource_name
+        self.content = content
+
+    def __str__(self):
```

Comment:
Hmm...

We could require `six` to be installed or if we don't want to require another library add [the code behind](https://bitbucket.org/gutworth/six/src/78315705481c3f5e491a3816e41aa66c54c8f965/six.py?at=default#cl-803) `python_2_unicode_compatible` to the utils file.